includes/class-amp-reader-themes.php

```diff
@@ -0,0 +1,546 @@
+		// the amp_reader_themes filter will show on the reader themes screen but will need to be manually installed on
+		// the site.
+		$default_reader_theme_slugs = wp_list_pluck( $this->get_default_raw_reader_themes(), 'slug' );
+		if ( ! in_array( $theme['slug'], $default_reader_theme_slugs, true ) ) {
+			return false;
+		}
+
+		if ( is_null( $this->can_install_themes ) ) {
+			if ( ! class_exists( 'WP_Upgrader' ) ) {
+				require_once ABSPATH . 'wp-admin/includes/file.php';
+				require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+			}
+
+			$this->can_install_themes = true === ( new WP_Upgrader() )->fs_connect( [ get_theme_root() ] );
+		}
+
+		if ( ! $this->can_install_themes ) {
+			return false;
+		}
+
+		if ( ! empty( $theme['requires'] ) && ! version_compare( get_bloginfo( 'version' ), $theme['requires'], '>=' ) ) {
+			return false;
+		}
+
+		if ( ! empty( $theme['requires_php'] ) && ! version_compare( phpversion(), $theme['requires_php'], '>=' ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Returns reader theme availability status.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param array $theme Theme data.
+	 * @return string Theme availability status.
+	 */
+	public function get_theme_availability( $theme ) {
+		switch ( true ) {
+			case $this->get_current_theme_name() === $theme['name']:
+				return self::ACTIVE_STATUS;
+
+			case wp_get_theme( $theme['slug'] )->exists():
+				return self::INSTALLED_STATUS;
+
+			case $this->can_install_theme( $theme ):
+				return self::INSTALLABLE_STATUS;
+
+			default:
+				return self::NON_INSTALLABLE_STATUS;
+		}
+	}
+
+	/**
+	 * Provides details for the classic theme included with the plugin.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @return array
+	 */
+	private function get_classic_mode() {
+		return [
+			'name'           => 'AMP Classic',
+			'slug'           => 'classic',
+			'preview_url'    => 'https://amp-wp.org',
+			'screenshot_url' => '//via.placeholder.com/218x472',
+			'homepage'       => 'https://amp-wp.org',
+			'description'    => __(
+				// @todo Improved description text.
+				'A legacy default template that looks nice and clean, with a good balance between ease and extensibility when it comes to customization.',
+				'amp'
+			),
+			'requires'       => false,
+			'requires_php'   => false,
+			'download_link'  => '',
+			'availability'   => [
+				'is_active'         => false,
+				'can_install'       => true,
+				'is_compatible_wp'  => true,
+				'is_compatible_php' => true,
+				'is_installed'      => true,
+			],
+		];
+	}
+
+	/**
+	 * Provides raw data for the default reader themes. Descriptions are translatable.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @return array
+	 */
+	public static function get_default_raw_reader_themes() {
+		// Copied from JSON data returned from Wordpress.org themes endpoint. See ::get_default_supported_reader_themes.
+		// Note: Descriptions are shortened and made translatable in the AMP_Reader_Theme_REST_Controller.
+		return [
```

Comment:
This array will be constructed in dynamic memory each time it is requested.
As it is completely static, it should be put into a class constant instead.

This would then turn `AMP_Reader_Themes::get_default_raw_reader_themes()` into `AMP_Reader_Themes::DEFAULT_THEMES` or so.